v2/googleads-to-bigquery/src/main/java/com/google/cloud/teleport/v2/io/GoogleAdsIO.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.io;
+
+import com.google.ads.googleads.v10.errors.GoogleAdsException;
+import com.google.ads.googleads.v10.errors.GoogleAdsFailure;
+import com.google.ads.googleads.v10.errors.QuotaErrorDetails;
+import com.google.ads.googleads.v10.errors.QuotaErrorEnum;
+import com.google.ads.googleads.v10.services.GoogleAdsRow;
+import com.google.ads.googleads.v10.services.GoogleAdsServiceClient;
+import com.google.ads.googleads.v10.services.SearchGoogleAdsStreamRequest;
+import com.google.ads.googleads.v10.services.SearchGoogleAdsStreamResponse;
+import com.google.auto.value.AutoValue;
+import com.google.cloud.teleport.v2.options.GoogleAdsOptions;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.List;
+import org.apache.beam.sdk.coders.BigEndianLongCoder;
+import org.apache.beam.sdk.coders.ListCoder;
+import org.apache.beam.sdk.extensions.protobuf.ProtoMessageSchema;
+import org.apache.beam.sdk.metrics.Counter;
+import org.apache.beam.sdk.metrics.Metrics;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.options.ValueProvider.NestedValueProvider;
+import org.apache.beam.sdk.options.ValueProvider.StaticValueProvider;
+import org.apache.beam.sdk.schemas.SchemaCoder;
+import org.apache.beam.sdk.transforms.*;
+import org.apache.beam.sdk.transforms.display.DisplayData;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.beam.sdk.values.TypeDescriptors;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class GoogleAdsIO {
+  public static Read read() {
+    return new AutoValue_GoogleAdsIO_Read.Builder()
+        .setGoogleAdsClientFactory(DefaultGoogleAdsClientFactory.getInstance())
+        .setRateLimitPolicyFactory(RateLimitPolicyFactory.withoutRatelimiter())
+        .setInferBeamSchema(false)
+        .build();
+  }
+
+  public static ReadAll readAll() {
+    return new AutoValue_GoogleAdsIO_ReadAll.Builder()
+        .setGoogleAdsClientFactory(DefaultGoogleAdsClientFactory.getInstance())
+        .setRateLimitPolicyFactory(RateLimitPolicyFactory.withoutRatelimiter())
+        .setInferBeamSchema(false)
+        .build();
+  }
+
```

Comment:
Since this IO may eventually be added to Beam for others to use, I would recommend adding extensive javador in the file header and for the `read` and `readAll` functions.